### PR TITLE
Added fully working TCP reconnect with platform connecting and disconnecting

### PR DESCRIPTION
**Pull request by nexylmao**

### CODE REVIEW

File: src/modbus/ModbusClient.cpp [hunk 1]
```diff
@@ -40,12 +44,16 @@ bool ModbusClient::connect()
...
 
-    if (modbus_connect(m_modbus) == -1)
-    {
+    LOG(ERROR) << "LibModbusClient: Attempting to connect";
+    while (modbus_connect(m_modbus) == -1) {
         LOG(ERROR) << "LibModbusClient: Unable to connect - " << modbus_strerror(errno);
-        destroyContext();
-        return false;
+        std::this_thread::sleep_for(std::chrono::seconds(m_timeoutDurations[m_timeoutIterator]));
+        if (m_timeoutIterator < 10) {
```
[the remark this answers is not in the document]
Comment (djesic) [reply src/modbus/ModbusClient.cpp]: I just saw that you changed the array, still wouldn't the iterator 10 be out the array?

[the remark this answers is not in the document]
Comment (nexylmao) [reply src/modbus/ModbusClient.cpp]: Timing changes is a later commit, and it is 10 members in length

[the remark this answers is not in the document]
Comment (djesic) [reply src/modbus/ModbusClient.cpp]: So max iterator value should be 9 then

[the remark this answers is not in the document]
Comment (nexylmao) [reply src/modbus/ModbusClient.cpp]: Yeah, you’re right. I wrote it like it’s a for loop...

**Pull request merged**